File: src/main/java/group/yunxin/service/ExamService.java
package group.yunxin.service;
import java.util.List;
import group.yunxin.pojo.TbExam;

import entity.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface ExamService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbExam> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbExam exam);
	
	
	/**
	 * 修改
	 */
	public void update(TbExam exam);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbExam findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbExam exam, int pageNum,int pageSize);
	
}

File: src/main/resources/static/js/service/examService.js
//服务层
app.service('examService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('/exam/findAll');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('/exam/findPage?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('/exam/findOne?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('/exam/add',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('/exam/update',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('/exam/delete?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('/exam/search?page='+page+"&rows="+rows, searchEntity);
	}    	
});
